packages/server/modules/core/utils/chunking.ts

```diff
@@ -0,0 +1,62 @@
+import { BaseError } from '@/modules/shared/errors'
+import { Options } from 'verror'
+
+type InsertionObject = {
+  data: string
+}
+
+export class ArgumentError extends BaseError {
+  static defaultMessage = 'Invalid argument value provided'
+
+  constructor(message?: string | undefined, options?: Options | Error | undefined) {
+    super(message, options)
+  }
+}
+
+// Js uses utf16 so string size in byes is length * 2
+export const calculateStringByteSize = (str: string) => new Blob([str]).size
```

Comment:
Note quick discussion with Gergo: does this "copy" the string? If so, are we making it worse? If yes, we could fall back to a napkin calculation based on the string length...